add migrate from eslint (#775)
oxc-project/oxc-project.github.io: src/docs/guide/usage/linter/migrate-from-eslint.md

## Running Oxlint and ESLint together

If not all required rules are available in Oxlint, you can run Oxlint and ESLint side by side.

A common setup is:

1. Enable Oxlint for all supported rules
2. Keep ESLint for unsupported rules
3. Disable overlapping rules in ESLint

Because Oxlint is significantly faster than ESLint, it is recommended to run Oxlint first to catch errors early, then fall back to ESLint only if needed.

For example:

```bash
oxlint && eslint
```

### Disabling overlapping rules in ESLint

You can use [`eslint-plugin-oxlint`](https://www.npmjs.com/package/eslint-plugin-oxlint) to disable ESLint rules that are already handled by Oxlint:

```bash
npm install --save-dev oxlint-eslint-plugin
This reduces duplicate diagnostics, can help cut down your linting time considerably, and allows ESLint to focus only on rules that Oxlint does not yet support.

Long-term - once remaining important rules been added in Oxlint - we strongly recommend moving fully to Oxlint to simplify your setup and reduce the number of npm dependencies for your project.

> ## Migrating from legacy ESLint configs

Reviewer: ```suggestion
## Migrating from legacy ESLint (v8.x) configs
```

If your project uses legacy ESLint config files (such as `.eslintrc.js` or `.eslintrc.json`), they must be migrated manually.

> The "legacy" ESLint v8.x configuration file shape maps closely to Oxlint’s config format, so most rules and options can be translated directly.

Reviewer: Not going to apply this myself because I'm not 100% sure this is safe/smart to do, but making it more explicit that you can potentially just copy-paste:

```suggestion
The "legacy" ESLint v8.x configuration file shape maps closely to Oxlint’s config format, so most rules and options can be translated directly. You may consider copy-pasting the ESLint v8 config file into an `.oxlintrc.json` file and then working from there. 

Note that after doing this, you should also add the `$schema` field for better editor integration/diagnostics, to help understand what is/isn't valid in an `.oxlintrc.json`: `"$schema": "./node_modules/oxlint/configuration_schema.json"`
```

Reviewer 2: Only if you use ESLint rules, other rules require `plugins` (rust rules) or `jsPlugin`, which is not 1:1 what ESLint does. 
I think for our rust Plugins, we deny `eslint-plugin-` prefix and ESLint expect them (not 100% on this one).

> For simpler configurations, you may also migrate them to an ESLint flat config first, and _then_ to Oxlint using `@oxlint/migrate`.

Reviewer: Unsure if we should necessarily link to the ESLint migrator, as it's kinda... mid, but:

```suggestion
For simpler legacy configuration files, you may also [migrate them to an ESLint flat config](https://www.npmjs.com/package/@eslint/migrate-config) first, and _then_ to Oxlint using `@oxlint/migrate`.
```

Reviewer 2: I think this is the new way intended by the ESLint team, other way is manually with https://www.npmjs.com/package/@eslint/eslintrc

If your project uses legacy ESLint config files (such as `.eslintrc.js` or `.eslintrc.json`), they must be migrated manually.

The "legacy" ESLint v8.x configuration file shape maps closely to Oxlint’s config format, so most rules and options can be translated directly.

For simpler legacy configuration files, you may also [migrate them to an ESLint flat config](https://www.npmjs.com/package/@eslint/migrate-config) first, and _then_ to Oxlint using `@oxlint/migrate`.

## Rule/plugin support

You may have specific rules that you rely on in ESLint that are not yet ported to Oxlint.

Almost all rules from our supported plugins will be ported - and a majority already have been. For those that will not be ported, some rules are deprecated in the original plugins, or have alternatives implemented already.

You can check the [meta issue](https://github.com/oxc-project/oxc/issues/481) for rule/plugin implementation status to see if the rules you rely on are planned for implementation, or if they have already been implemented by other, equivalent rules.

For plugins that are not implemented natively in Oxlint, it is recommended to use JS Plugins.

## Next steps

> - [Setup editors](/docs/guide/usage/linter/editors)

Reviewer 2: ```suggestion
```

Is this really needed? Other pages do not have a "next steps" but a pagination for the page navigation

Reviewer: Some pages do have the `Next steps` section, and I think it may be worth including the editors link on this page somewhere, but it is a bit of a weird way to do this. I'll remove it for now.

# Migrate from ESLint

> This guide is for existing JavaScript or TypeScript projects that currently use ESLint and want to migrate to Oxlint.

Reviewer: ```suggestion
This guide is for existing JavaScript and TypeScript projects that currently use ESLint and want to migrate to Oxlint.
```

If you are starting a new project, or do not need to preserve an existing ESLint configuration, we recommend initializing a fresh Oxlint config instead:

```bash
oxlint --init
```

## Overview

Oxlint and ESLint share similar configuration concepts, but they differ in supported rules and config formats.

When migrating, expect the following:

- Most ESLint core rules and popular plugin rules are supported
- Some rules may not yet be available
- ESLint configuration files must be converted to Oxlint’s config format
- Oxlint is designed for incremental adoption; a full migration is not required upfront

## Migrating from ESLint Flat Config

> If your project uses ESLint Flat Config (e.g. `eslint.config.js`), you can migrate automatically using [`@oxlint/migrate`](https://www.npmjs.com/package/@oxlint/migrate).

Reviewer: ```suggestion
If your project uses an ESLint flat file config (e.g. `eslint.config.js` or `eslint.config.mjs`), you can migrate automatically using [`@oxlint/migrate`](https://www.npmjs.com/package/@oxlint/migrate).
```

### Run the migration tool

From the project root:

```bash
npx @oxlint/migrate <optional-eslint-flat-config-path>
```

This command:

> - Reads your ESLint Flat Config

Reviewer: ```suggestion
- Reads your ESLint flat file config
```

- Converts supported rules to an Oxlint config
- Preserves rule severities and options where possible

The generated config can be edited manually after migration.

> See also the list of [available options](https://github.com/oxc-project/oxlint-migrate?tab=readme-ov-file#options).

Reviewer: ```suggestion
See the list of [available options](https://github.com/oxc-project/oxlint-migrate?tab=readme-ov-file#options) for more details.
```

### Type-aware TypeScript rules

If your ESLint setup uses `typescript-eslint` with type-aware rules, pass the `--type-aware` flag:

```bash
npx @oxlint/migrate --type-aware
```

> This ensures the generated Oxlint config enables the required settings for type-aware analysis.

Reviewer: ```suggestion
This ensures the generated Oxlint config includes type-aware rules.

Note that type-aware support is in alpha. For further information on Oxlint's type-aware support, see [the Type-Aware Linting page](docs/guide/usage/linter/type-aware).
```

### JavaScript plugins

If your ESLint config uses plugins that are not supported natively by Oxlint, you can retain them using JavaScript plugins:

```bash
npx @oxlint/migrate --js-plugins
```

> This allows you to continue using those rules while migrating the rest of your configuration.

Reviewer: ```suggestion
This allows you to continue using those rules while migrating the rest of your configuration. For more information on JavaScript Plugins, see [the JS Plugins page](docs/guide/usage/linter/js-plugins).
```

## Running Oxlint and ESLint together

If not all required rules are available in Oxlint, you can run Oxlint and ESLint side by side.

A common setup is:

1. Enable Oxlint for all supported rules
2. Keep ESLint for unsupported rules
3. Disable overlapping rules in ESLint

Because Oxlint is significantly faster than ESLint, it is recommended to run Oxlint first to catch errors early, then fall back to ESLint only if needed.

For example:

```bash

> oxlint && eslint

Reviewer: ```suggestion
npx oxlint && npx eslint
```

```

### Disabling overlapping rules in ESLint

You can use [`eslint-plugin-oxlint`](https://www.npmjs.com/package/eslint-plugin-oxlint) to disable ESLint rules that are already handled by Oxlint:

```bash
npm install --save-dev oxlint-eslint-plugin
```

> 

Reviewer: ````suggestion
```bash
npm install --save-dev oxlint-eslint-plugin
```

And then update your ESLint config with the plugin. See [the docs](https://github.com/oxc-project/eslint-plugin-oxlint?tab=readme-ov-file#usage) for more information.
````

> This reduces duplicate diagnostics and allows ESLint to focus only on rules that Oxlint does not yet support.

Reviewer: ```suggestion
This reduces duplicate diagnostics, can help cut down your linting time considerably, and allows ESLint to focus only on rules that Oxlint does not yet support.

Long-term - once remaining important rules been added in Oxlint - we strongly recommend moving fully to Oxlint to simplify your setup and reduce the number of npm dependencies for your project.
```
